doc/source/Code-Style-Guide.rst

```diff
@@ -0,0 +1,746 @@
+within the class; as a static method if it makes no use of self.
+
+Command Line Interface (CLI)
+----------------------------
+
+Add a command line interface (improver/cli/.py) to invoke plugins that
+can be used as a standalone utility or executable within a suite context
+(e.g. wind downscaling, neighbourhood processing, spot data extraction).
+These CLIs are invoked using ``bin/improver <cli-name>`` (note that the
+CLI filename uses underscores, but the call to use the CLI uses hyphens)
+
+IMPROVER CLIs should only have ``from improver import cli`` as the top
+level imports. Other imports are placed inside the function that uses
+them. This gives the benefit of a more rapid response to the command
+``bin/improver <cli-name> -h`` when those other (often slow) imports are
+not needed.
+
+Each CLI should have a process function. This will require a
+``@cli.clizefy`` decorator to gain the functionality of
+`clize <https://clize.readthedocs.io/en/stable/>`__. If you want the CLI
+to save a cube to disk, it will need the decorator ``@cli.with_output``,
+this will mean on the command line, the ``--output`` flag can be used to
+specify an output path.
+
+To load the cubes, each cube argument will need a type. For a basic cube
+this will be ``cube: cli.inputcube``. If there is a default argument to
+go with the typed variable, spaces are required around the ``=`` for
+example ``weights: cli.inputcube = None``. There are other types which
+can be used such as: - the python standards - ``float`` - ``int`` -
+``bool`` - specific additions - ``cli.inputcube`` - Where a string is
+given which is a path to a cube to load - ``cli.inputjson`` - Where a
+string is given which is a path to a json file to load -
+``cli.comma_separated_list`` - This will convert the argument into that
+format and deal with error handling if no conversion is possible.
+
+A complete list of local added variable types can be found by
+identifying all the ``@value_converter`` decorated functions in
+`cli/init.py <https://github.com/metoppv/improver/blob/master/improver/cli/__init__.py>`__.
+
+Arguments into the process function should start with the cubes. After
+all the cubes, there should be an argument of ``*``, this separates the
+positional arguments from the keyword arguments. If you are loading a
+cube list of unspecified number of cubes ``*cubelist`` will take all the
+cubes, load them and return them as a tuple.
+
+All arguments after the ``*`` will need to be given with keywords on the
+command line.
+
+Due to the use of ``*``, required arguments can be used before and after
+the star. for example
+
+.. code:: python
+
+   from improver import cli
+   @cli.clizefy
+   def process(cube: cli.inputcube,
+               weights: cli.inputcube = None,
+               *,
+               coord_for_masking,
+               radius: float = None)
+
+The required arguments in this example are: - cube - coord_for_masking
```

Comment:
Another bulleted list that has been inlined.

Reply:
Fixed.